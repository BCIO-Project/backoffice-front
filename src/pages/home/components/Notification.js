import React, { Component } from 'react';
import { Checkbox, Icon } from 'semantic-ui-react';
import * as notificationsService from './../../../services/notifications.service';

export default class Notification extends Component {
    /**
     * manage checkbox value
     * @memberof Notification
     * @param {object} e - onChange event in input
     * @param {object} data - input's information
     */
    handleReadStatus(e, data){
        //if checked or unchecked define action
        const checked = data.checked ? 'read' : 'unread';
        return checked
    }
    
    /**
     * handle read/unread notifications value
     * @memberof Notification
     */
    handleReadChange = (e, data) => {
        notificationsService.save(`${this.props.notificationId}/${this.handleReadStatus(e, data)}`)
        .then(() => {
            // fetch and update notifications
            this.props.getNotifications()
                      
        })
    }

    /**
     * delete notifications
     * @memberof Notification
     */
    handleEraseNotification = () => {
        notificationsService.erase(`${this.props.notificationId}`)
        .then(() => {
            // fetch and update notifications
            this.props.getNotifications()
        })
    }
    render() {
        const { text, read, page, position } = this.props
        return (
            <div className='notification_info'>
                <div className="campaign_category">
                    <p className="campaign_category_text">
                        {`${page} / ${position}`}
                    </p>
                    <div>
                        <Checkbox label='Read' onChange={this.handleReadChange} checked={read}/>
                        <button className='notifications_erase' onClick={this.handleEraseNotification}><Icon name='close'/></button>
                    </div>
                </div>
                <p className='notifications_text'>{text}</p>
            </div>
        )
    }
}
