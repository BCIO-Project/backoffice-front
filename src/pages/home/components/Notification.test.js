import React from 'react';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Notification from './Notification';

describe('<Notification>', ()=>{
    let notification;
    beforeEach(()=>{
        notification = shallow(<Notification></Notification>)
    })
    test('renders', ()=>{
        expect(notification.exists()).toBe(true);
    })
    test('Should render correctly', () =>{
        const component = renderer.create(<Notification />)
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
    test('should execute function onClick', ()=>{
        const instance = notification.instance();
        const handleEraseNotificationSpy = jest.spyOn(instance, 'handleEraseNotification');
        instance.forceUpdate();
        notification.find('.notifications_erase').simulate('click')
        expect(handleEraseNotificationSpy).toHaveBeenCalled();
    })
    it('should return read if checked', ()=>{
        const onChangeMockEvent = {} 
        const onChangeMockData = {checked: true}  
        expect(notification.instance().handleReadStatus(onChangeMockEvent, onChangeMockData)).toBe('read')
    })
})