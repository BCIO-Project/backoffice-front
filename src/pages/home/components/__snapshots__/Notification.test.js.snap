// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification> Should render correctly 1`] = `
<div
  className="notification_info"
>
  <div
    className="campaign_category"
  >
    <p
      className="campaign_category_text"
    >
      undefined / undefined
    </p>
    <div>
      <div
        className="ui checkbox"
        onChange={[Function]}
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <input
          checked={false}
          className="hidden"
          readOnly={true}
          tabIndex={0}
          type="checkbox"
        />
        <label>
          Read
        </label>
      </div>
      <button
        className="notifications_erase"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="close icon"
          onClick={[Function]}
        />
      </button>
    </div>
  </div>
  <p
    className="notifications_text"
  />
</div>
`;
